src/actions/definitionActions.js

```diff
@@ -96,14 +96,15 @@ export function getDefinitionSuggestedDataAction(token, prefix, name) {
   }
 }
 
-export function browseDefinitionsAction(token, entity, name) {
+export function browseDefinitionsAction(token, query, name) {
   return async dispatch => {
     const actions = asyncActions(name)
     dispatch(actions.start())
     try {
-      const result = await browseDefinitions(token, entity)
-      dispatch(actions.success({ add: result }))
-      const toAdd = map(result, component => EntitySpec.validateAndCreate(component.coordinates)).filter(e => e)
+      const result = await browseDefinitions(token, query)
+      const definitions = result.data
```

Comment:
@storrisi the other property here is 

```js
const continuationToken = result.continuationToken
```

if you add this to the query it will give you the next page of results. We should call when we get near to the bottom of the list scrolling. The continuationToken should be removed from the query when any filter has changed in the UI so we start back at page 1.